src/olympia/blocklist/cron.py

```diff
@@ -119,28 +116,25 @@ def _upload_mlbf_to_remote_settings(*, force_base=False):
         len(mlbf.data.not_blocked_items),
     )
 
-    make_base_filter = (
-        force_base
-        or base_filter is None
-        or previous_filter is None
-        or mlbf.blocks_changed_since_previous(BlockType.BLOCKED, base_filter)
-        > BASE_REPLACE_THRESHOLD
-    )
+    # Until we are ready to enable soft blocking, it should not be possible
+    # to create a stash and a filter at the same iteration
+    if create_stash and len(base_filters_to_update) > 0:
+        raise Exception(
+            'Cannot upload stash and filter without implementing soft blocking'
+        )
 
-    if make_base_filter:
-        mlbf.generate_and_write_filter()
-    else:
+    if create_stash:
         mlbf.generate_and_write_stash(previous_filter)
 
+    for block_type in base_filters_to_update:
+        mlbf.generate_and_write_filter(block_type)
+
     upload_filter.delay(
-        generation_time,
-        filter_list=[BlockType.BLOCKED.name] if make_base_filter else [],
-        create_stash=not make_base_filter,
+        mlbf.created_at,
```

Comment:
I might have missed something. Why do we change the date here?

Reply:
They have the same value. In the original PR, I switched to using the creation date because this is the value that is actually assigned to the MLBF class. It's also the value we're passing to the initialize but it just makes sense to me to use the value from the class.